### PR TITLE
add sensor graph sample

### PR DESCRIPTION
**Pull request by proppy**
Sensor graph is a C++ Android sample that read current accelerometer values and draw them using OpenGL.

It demonstrate usage of the following Native C++ API:
- [Sensors](http://developer.android.com/ndk/reference/group___sensor.html)
- [Assets](http://developer.android.com/ndk/reference/group___asset.html)

screenshot

/cc @ggfan @hak

### CODE REVIEW

File: sensor-graph/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,45 @@
+apply plugin: 'com.android.model.application'
+
+model {
+    android {
+        compileSdkVersion = 21
+        buildToolsVersion = "22.0.1"
+
+        defaultConfig.with {
+            applicationId    = "com.android.gl2jni"
```
Comment (proppy) [inline sensor-graph/app/build.gradle]: s/gl2jni/sensorgraph

File: sensor-graph/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+*/
+-->
+
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.android.sensorgraph">
+    <uses-feature android:glEsVersion="0x00020000" android:required="true"/>
+    <application
+            android:icon="@mipmap/ic_launcher"
+            android:allowBackup="false"
+            android:label="@string/gl2jni_activity">
```
Comment (proppy) [inline sensor-graph/app/src/main/AndroidManifest.xml]: s/gl2jni/sensorgraph/

File: sensor-graph/app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+** limitations under the License.
+*/
+-->
+
+<!-- This file contains resource definitions for displayed strings, allowing
+     them to be changed based on the locale and options. -->
+
+<resources>
+    <!-- Simple strings. -->
+    <string name="gl2jni_activity">GL2JNI</string>
```
Comment (proppy) [inline sensor-graph/app/src/main/res/values/strings.xml]: s/gl2jni/sensorgraph/

**Comment by proppy** [comment]
@ggfan @hak @claywilkinson @gguuss does someone have the bandwidth to review this?

**Comment by ggfan** [comment]
just a few comment:
1) I feel the program leaks the shader object memory [vertex and pixel shader object]
2) the native module name could be something else, more relate to sensor rather than gl2jni
3) is java support egl1.4, maybe use it to replace egl1.0, just an comment
thank you for creating the sample!

File: sensor-graph/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (proppy) [inline sensor-graph/app/src/main/AndroidManifest.xml]: Done.

File: sensor-graph/app/src/main/res/values/strings.xml (hunk 3 above)
Comment (proppy) [inline sensor-graph/app/src/main/res/values/strings.xml]: Done.

**Comment by proppy** [comment]
@ggfan feel free to comment directly on the code too.

> 2) the native module name could be something else, more relate to sensor rather than gl2jni
Done.

**Comment by proppy** [comment]
> 1) I feel the program leaks the shader object memory [vertex and pixel shader object]

You mean I should delete them with glDeleteShader after calling glLinkProgram? I think hello-gl2 has also this problem, we should probably fix it too.

**Comment by proppy** [comment]
> 3) is java support egl1.4, maybe use it to replace egl1.0, just an comment

I'm just using the default `GLSurfaceView` with a GLES 2.0 context, I'm not sure how to force GLSurfaceView to use egl1.4, do you have an example?

File: sensor-graph/app/build.gradle (hunk 1 above)
Comment (proppy) [inline sensor-graph/app/build.gradle]: Done.

**Comment by proppy** [comment]
> 1) I feel the program leaks the shader object memory [vertex and pixel shader object]

@ggfan Done.

**Comment by proppy** [comment]
PTAL

**Comment by ggfan** [comment]
LGTM

**Comment by proppy** [comment]
Travis is green, merging.

**Pull request merged**